src/pgvecto_rs/types/svector.py

```diff
@@ -0,0 +1,207 @@
+    ):
+        return cls._from_parts(dim, [v for v in indices], [v for v in values])
+
+    def __repr__(self):
+        elements = dict(zip(self._indices, self._values))
+        return f"SparseVector({elements}, {self._dim})"
+
+    def dimensions(self):
+        return self._dim
+
+    def indices(self):
+        return self._indices
+
+    def values(self):
+        return self._values
+
+    def to_coo(self):
+        from scipy.sparse import coo_array
+
+        coords = ([0] * len(self._indices), self._indices)
+        return coo_array((self._values, coords), shape=(1, self._dim))
+
+    def to_list(self):
+        vec = [0.0] * self._dim
+        for i, v in zip(self._indices, self._values):
+            vec[i] = v
+        return vec
+
+    def to_numpy(self):
+        vec = np.repeat(0.0, self._dim).astype(np.float32)
+        for i, v in zip(self._indices, self._values):
+            vec[i] = v
+        return vec
+
+    def to_text(self):
+        return (
+            "{"
+            + ",".join(
+                [f"{int(i)}:{float(v)}" for i, v in zip(self._indices, self._values)]
+            )
+            + "}/"
+            + str(int(self._dim))
+        )
+
+    def to_binary(self):
+        # convert indices to little-endian uint32
+        indices = np.asarray(self._indices, dtype="<I")
+        indices_len = indices.shape[0]
+        indices_bytes = indices.tobytes()
+        # convert values to little-endian float32
+        values = np.asarray(self._values, dtype="<f")
+        values_len = values.shape[0]
+        values_bytes = values.tobytes()
+        # check indices and values length is the same
+        if indices_len != values_len:
+            raise ValueError(
+                "sparse vector expected indices length %d to match values length %d"
+                % (indices_len, values_len)
+            )
+        return (
+            pack("<I", self._dim)
+            + pack("<I", indices_len)
+            + indices_bytes
+            + values_bytes
+        )
+
+    def _from_dict(self, d, dim):
+        elements = [(i, v) for i, v in d.items() if v != 0]
+        elements.sort()
+
+        self._dim = int(dim)
+        self._indices = [int(v[0]) for v in elements]
+        self._values = [float(v[1]) for v in elements]
+
+    def _from_sparse(self, value):
+        value = value.tocoo()
+
+        if value.ndim == 1:
+            self._dim = value.shape[0]
+        elif value.ndim == 2 and value.shape[0] == 1:  # noqa: PLR2004
+            self._dim = value.shape[1]
+        else:
+            raise SparseShapeError(value.shape)
+
+        if hasattr(value, "coords"):
+            # scipy > 1.13
+            self._indices = value.coords[0].tolist()
+        else:
+            self._indices = value.col.tolist()
+        self._values = value.data.tolist()
+
+    def _from_dense(self, value):
+        self._dim = len(value)
+        self._indices = [i for i, v in enumerate(value) if v != 0]
```

Comment:
I guess `v` is float, so it's not a good idea to compare it with 0 directly. Try `math.isclose` or `np.isclose`.